File: SARMobileClient/app/src/main/java/se3810/groupi/Tag.kt
package se3810.groupi

class Tag(id: Int, loc: Location){
    var id : Int
    var location : Location
    var label : ArrayList<String>
    var image: ArrayList<String>
    var location_id : Int
    var users_id : Int

    init {
        this.id = id
        this.location = loc
        this.label = ArrayList<String>()
        this.image = ArrayList<String>()
        this.location_id = 1
        this.users_id = 1;
    }

    /**
     * Add a text label to be displayed with the tag
     * label: the text to store as a text label
     */
    fun addTextLabel(newLabel : String) {
        label.add(newLabel)
    }

    /**
     * Add an image label to be displayed with the tag
     * img: the image to store as an image label
     */
    fun addImageLabel(img : String) {
        image.add(img)
    }
}

File: sar_server/app/models/tag.rb
# frozen_string_literal: true

# The displayed content of the SAR application
# @!attribute image - an image stored as raw text
# @!attribute label -
class Tag < ApplicationRecord
  belongs_to :location, foreign_key: :location_id
  belongs_to :user, foreign_key: :users_id

  def self.spoofed_create(params)
    params[:users_id] = User.first.id
    if params[:location_id].nil?
      params[:location_id] =
        Location.find_or_create_by(latitude: params[:latitude], longitude: params[:longitude]).id
    end
    Tag.create(image: params[:image],
               label: params[:label],
               users_id: params[:users_id],
               location_id: params[:location_id])
  end

  def self.near_me(origin, magnitude)
    locations_with_tags = Location.all_with_tags(origin, magnitude)
    tags = []
    locations_with_tags.each do |loc|
      loc.tags.each do |t|
        tags << t
      end
    end
    tags
  end
end
